### PR TITLE
feat: doc support header and footer

### PR DESCRIPTION
**Pull request by Jocs**
close https://github.com/dream-num/univer-pro/issues/1307

## Pull Request Checklist

- [x] Related tickets or issues have been linked in the PR description (or missing issue).
- [x] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [x] Unit tests have been added for the changes (if applicable).
- [x] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Comment by DR-Univer** [comment]
image
sheet editor 左上角和右上角有残留得header图标。

**Comment by DR-Univer** [comment]
image
普通editor也有

File: packages/core/src/types/interfaces/i-document-data.ts [hunk 1]
```diff
@@ -301,14 +301,19 @@ export enum BlockType {
...
-    defaultFooterId?: string; // defaultFooterId
+    defaultHeaderId?: string; // defaultHeaderId or the odd page header id
+    defaultFooterId?: string; // defaultFooterId or the odd page footer id
     evenPageHeaderId?: string; // evenPageHeaderId
     evenPageFooterId?: string; // evenPageFooterId
     firstPageHeaderId?: string; // firstPageHeaderId
     firstPageFooterId?: string; // firstPageFooterId
     useFirstPageHeaderFooter?: BooleanNumber; // useFirstPageHeaderFooter
-    useEvenPageHeaderFooter?: BooleanNumber; // useEvenPageHeaderFooter,
+    evenAndOddHeaders?: BooleanNumber; // useEvenPageHeaderFooter,
```
[the remark this answers is not in the document]
Comment (Jocs) [reply packages/core/src/types/interfaces/i-document-data.ts:333]: follow XML 标准中的命名。但是useFirstPageHeaderFooter延续了之前的命名

File: packages/core/src/types/interfaces/i-selection-data.ts [hunk 2]
```diff
@@ -70,6 +70,7 @@ export interface ITextRange extends ITextRangeStart {
 
 export interface ITextRangeParam extends ITextRange {
     segmentId?: string; //The ID of the header, footer or footnote the location is in. An empty segment ID signifies the document's body.
+    segmentPage?: number; //The page number of the header, footer or footnote the location is in. An empty segment ID signifies the document's body.
```
[the remark this answers is not in the document]
Comment (Jocs) [reply packages/core/src/types/interfaces/i-selection-data.ts:77]: 会更新的。

**Review by DR-Univer — APPROVED** [review]

**Pull request merged**